pe/orchestration.py

```diff
@@ -0,0 +1,246 @@
+        next_params: Union[Dict] = canvas_params
+        LOGGER.debug(f'Params for first request: {next_params}')
+
+        while more_pages:
+            LOGGER.debug(f'Page number {page_num}')
+            response: Union[Response, None] = api_call_with_retries(
+                self.api_handler,
+                get_subs_url,
+                CANVAS_SCOPE,
+                'GET',
+                next_params
+            )
+            if response is None:
+                LOGGER.info('api_call_with_retries failed to get a response; no more data will be collected')
+                more_pages = False
+            else:
+                sub_dicts += json.loads(response.text)
+                page_info: Union[None, Dict[str, Any]] = self.api_handler.get_next_page(response)
+                if not page_info:
+                    more_pages = False
+                else:
+                    LOGGER.debug(f'Params for next page: {page_info}')
+                    next_params = page_info
+                    page_num += 1
+
+        LOGGER.info(f'Gathered {len(sub_dicts)} submissions from Canvas')
+        LOGGER.debug(sub_dicts)
+        return sub_dicts
+
+    def create_sub_records(self, sub_dicts: List[Dict[str, Any]]) -> None:
+        """
+        Parses Canvas submission records and writes them to the database.
+
+        :param sub_dicts: Dictionary results of Canvas API search in get_sub_dicts_for_exam
+        :type sub_dicts: List of dictionaries with string keys
+        """
+        if len(sub_dicts) == 0:
+            LOGGER.info('No sub_dicts were provided')
+        else:
+            try:
+                Submission.objects.bulk_create(
+                    objs=[
+                        Submission(
+                            submission_id=sub_dict['id'],
+                            exam=self.exam,
+                            student_uniqname=sub_dict['user']['login_id'],
+                            submitted_timestamp=sub_dict['submitted_at'],
+                            graded_timestamp=sub_dict['graded_at'],
+                            score=sub_dict['score'],
+                            transmitted=False
+                        )
+                        for sub_dict in sub_dicts
+                    ]
+                )
+                LOGGER.info(f'Inserted {len(sub_dicts)} new Submission records in the database')
+            except Exception as e:
+                LOGGER.error(e)
+                LOGGER.error('Submissions bulk creation failed')
+
+    def send_scores(self, subs_to_send: List[Submission]) -> None:
+        """
+        Send scores for exam with unique student_uniqname values in bulk to M-Pathways.
+
+        :param subs_to_send: List of un-transmitted Submission model instances
+        :type subs_to_send: List of Submission model instances
+        :return: None
+        :rtype: None
+        """
+        send_scores_url: str = 'aa/SpanishPlacementScores/Scores'
```

Comment:
Yeah, I can make this a constant of some sort. If it's always going to be the Pascal Case version of the scope, then yeah, some of that could be re-factored to avoid duplication. I'll work on this and think about it for the other URL.
I haven't heard back yet about if the new endpoint is ready. So to be clear, you don't think this should be configure-able?

Reply:
Okay, I think I did what you suggested here: https://github.com/tl-its-umich-edu/placement-exams/pull/12/commits/64c9750803572757612f29eef65df91b2b370a86

Reply:
I think I'm going to stick with what I have in the commit. Using `lower` won't change the number of imports, since I'd have to import `MPATHWAYS_SCOPE_CAPS` instead of `MPATHWAYS_SCOPE`, so I'd rather just call `lower` once in `constants.py`.